hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/HandleException.java

```diff
@@ -34,6 +34,11 @@ public HandleException(final ResponseCodeEnum status) {
         this.status = status;
     }
 
+    public HandleException(final ResponseCodeEnum status, Throwable originalCause) {
```

Comment:
This does not make any sense IMO, because `originalCause` will never be used. If you feel the need to store a cause with a `HandleException` you are probably doing something wrong.

Please remove this constructor.

Reply:
@netopyr that request came from @jsync-swirlds as you can see above and also from @hendrikebbers

Reply:
We have a number of places where we throw a HandleException in replacement of another exception.  It's always good practice to include the cause when that's done, so it can show up in diagnostics and audit records.  Otherwise we have the situation of hiding errors and operations has a much harder time tracking down unusual errors.

Reply:
I would **always** be careful with words like **always**. 😉

While I think, in general, you are right, `HandleException` (and `PreCheckException`) are unique because they are reserved for user errors and must not be logged or otherwise occupy resources. We could have agreed to return an error code instead of throwing an `Exception` in many places. Providing a constructor that takes a cause creates the illusion that this information will be available somehow, while in reality, it is thrown away. If somebody thinks the cause needs to be preserved, they have to consider alternatives, and omitting the constructor forces that person to do that.

Reply:
If we're never logging the `HandleException` (or `PreCheckException`), then we should make that a lot clearer.
It's not really an exception in that case; it's rather a tricky form of unconditional jump being used to exit a processing unit with an unsuccessful result code.
Perhaps some documentation in the `HandleException` class to make abundantly clear that this is not permitted to wrap other exceptions, nor be thrown in response to another exception without logging (because all exception data will be lost).  In fact I would recommend leaving this constructor in place, but having it fail with `UnsupportedException` in order to make explicit that there is a clear design choice made to not accept a cause.

`PreCheckException` is equally in need of the same documentation, as we already have places where we discard exception data because that "exception" is thrown instead of handling other Throwables without any diagnostic.  It's critical in this system that we do not hide any exceptional events from diagnostic data; that's a strong opportunity for serious issues to evade detection.

Reply:
Here is a proposed block to make this design decision clear:
```
    /**
     * <strong>Disallowed</strong> constructor of {@code HandleException}.
     * This {@link Exception} subclass is used as a form of unconditional jump, rather than a true
     * exception.  If another {@link Throwable} caused this exception to be thrown, then that other
     * throwable <strong>must</strong> be logged to appropriate diagnostics before the {@code HandleException}
     * is thrown.
     *
     * @param responseCode the {@link ResponseCodeEnum responseCode}.  This is ignored.
     * @param cause the {@link Throwable} that caused this exception.  This is ignored.
     * @throws UnsupportedOperationException always.  This constructor must not be called.
     */
    public HandleException(@NonNull final ResponseCodeEnum responseCode, @Nullable final Throwable cause) {
        throw new UnsupportedOperationException("HandleException must not chain a cause");
    }

    /**
     * {@inheritDoc}
     * This implementation prevents initializing a cause.  HandleException is a result code carrier and
     * must not have a cause.  If another {@link Throwable} caused this exception to be thrown, then that other
     * throwable <strong>must</strong> be logged to appropriate diagnostics before the {@code HandleException}
     * is thrown.
     * @throws UnsupportedOperationException always.  This method must not be called.
     */
    @Override
    public Throwable initCause(Throwable cause) {
        throw new UnsupportedOperationException("HandleException must not chain a cause");
    }


```

Reply:
It is a good idea to override `initCause()`. I would not add the constructor, though. Of course, nobody from us would do that, but I heard sometimes people call methods without reading the JavaDocs thoroughly. 🙂 If there is no constructor, nobody can call it accidentally.

`PreCheckException` and `HandleException` should only be called if there was an error by the user. In such a case, we are not interested in the details; only the `responseCode` matters. Are there cases where important information gets lost when we throw away the causing `Exception`? Maybe we should review such places.

Reply:
The purpose of adding the constructor is to make sure unit tests fail if someone calls it, providing a strong signal not to use it and also avoiding the problem of people thinking it should be added (not everyone will always know the design goals, this makes those very clear and concrete).

Reply:
@netopyr Any thoughts on the rationale presented above?

Reply:
I still think we should not add the public constructor. People will call it without reading the docs, and I would not want to rely on the hope that a unit test will test the particular error condition. Documenting this in the javadocs for the class and the overridden `initCause()` method are warning enough IMO. But if you think the constructor is needed, I can agree to a private one.

Reply:
Added the code as you discussed.